roles/common/tasks/check_existing.yml

```diff
@@ -1,28 +1,31 @@
 ---
 - name: Check for existing {{ deployment_type }} deployment
+  kubernetes.core.k8s_info:
+    kind: "{{ kind }}"
+    api_version: '{{ api_version }}'
+    name: "{{ ansible_operator_meta.name }}"
+    namespace: '{{ ansible_operator_meta.namespace }}'
+  register: existing_deployment
+
+- name: Set previous_version if deployment exists
   block:
-    - name: Check if {{ kind }} CR exists by name
-      kubernetes.core.k8s_info:
-        kind: "{{ kind }}"
-        api_version: '{{ api_version }}'
-        name: "{{ ansible_operator_meta.name }}"
-        namespace: '{{ ansible_operator_meta.namespace }}'
-      register: existing_deployment
 
-    - name: Set previous_version version based on {{ deployment_type }} CR version status
+    # addresses a pre-existing bug in 2.4 in which deployedVersion is not set in the status
+    - name: Refresh {{ deployment_type }} status if deployedVersion is undefined in existing_deployment status
+      ansible.builtin.include_role:
+        name: galaxy-status
```

Comment:
wouldn't this fail on fresh installs because installed_plugins is not yet defined?

Reply:
I was wrong about this, it is defined earlier on.  Route_host is not though.  We need to re-order 
main.yml

Reply:
@aknochow we need this diff to move the includes of these tasks to after when `route_host` is defined:

```diff
diff --git a/roles/common/tasks/main.yml b/roles/common/tasks/main.yml
index 3e7f40b..574bde9 100644
--- a/roles/common/tasks/main.yml
+++ b/roles/common/tasks/main.yml
@@ -12,10 +12,6 @@
     api_version: '{{ hostvars["localhost"]["inventory_file"].split("/")[4:6] | join("/")  }}'
     kind: '{{ hostvars["localhost"]["inventory_file"].split("/")[6]  }}'
 
-- name: Check for existing {{ deployment_type }} deployment
-  ansible.builtin.include_tasks: check_existing.yml
-  when: gating_version | length
-
 - name: Fail execution if image_pull_secret or image_pull_secrets are defined but as NoneType ('image_pull_secret[s]:')
   fail:
     msg: "If image_pull_secret or image_pull_secrets will not be used, their declarations should be removed."
@@ -62,6 +58,10 @@
       web_url: "https://{{ route_host }}"
   when: ingress_type | lower == 'route'
 
+- name: Check for existing {{ deployment_type }} deployment
+  ansible.builtin.include_tasks: check_existing.yml
+  when: gating_version | length
+
 - name: Configure Postgres Configuration Secret
   include_tasks: postgres_configuration.yml
 ```

Reply:
I opened a PR to your branch:
* https://github.com/aknochow/galaxy-operator/pull/2